src/Bridges/Constraint/semi_to_binary.jl

```diff
@@ -0,0 +1,191 @@
+    return
+end
+
+function MOI.get(model::MOI.ModelLike, attr::MOI.ConstraintPrimal,
+    bridge::SemiToBinaryBridge)
+    MOI.get(model, MOI.VariablePrimal(attr.N), bridge.variable_index)
+end
+
+function MOIB.added_constrained_variable_types(::Type{<:SemiToBinaryBridge{T, S}}) where {T, S}
+    return [(MOI.ZeroOne,)]
+end
+
+function MOIB.added_constraint_types(::Type{<:SemiToBinaryBridge{T, S}}) where {T, S<:MOI.Semicontinuous{T}}
+    return [
+            (MOI.ScalarAffineFunction{T}, MOI.LessThan{T}),
+            (MOI.ScalarAffineFunction{T}, MOI.GreaterThan{T}),
+            ]
+end
+
+function MOIB.added_constraint_types(::Type{<:SemiToBinaryBridge{T, S}}) where {T, S <: MOI.Semiinteger{T}}
+    return [
+            (MOI.ScalarAffineFunction{T}, MOI.LessThan{T}),
+            (MOI.ScalarAffineFunction{T}, MOI.GreaterThan{T}),
+            (MOI.SingleVariable, MOI.Integer),
+            ]
+end
+
+function concrete_bridge_type(::Type{<:SemiToBinaryBridge{T}},
+                              ::Type{MOI.SingleVariable},
+                              ::Type{S}) where {T, S<:SemiSets}
+    return SemiToBinaryBridge{T, S}
+end
+
+# Attributes, Bridge acting as a model
+
+function MOI.get(::SemiToBinaryBridge, ::MOI.NumberOfVariables)
+    return 1
+end
+
+function MOI.get(b::SemiToBinaryBridge, ::MOI.ListOfVariableIndices)
+    return [b.binary_variable_index]
+end
+
+function MOI.get(::SemiToBinaryBridge{T, S},
+    ::MOI.NumberOfConstraints{MOI.SingleVariable, MOI.ZeroOne}) where {T, S}
+    return 1
+end
+
+function MOI.get(::SemiToBinaryBridge{T, S},
+    ::MOI.NumberOfConstraints{MOI.SingleVariable, MOI.Integer}) where {T, S<:MOI.Semiinteger}
+    return 1
+end
+
+function MOI.get(::SemiToBinaryBridge{T, S},
+    ::MOI.NumberOfConstraints{MOI.ScalarAffineFunction{T}, MOI.GreaterThan{T}}) where {T, S}
+    return 1
+end
+
+function MOI.get(::SemiToBinaryBridge{T, S},
+    ::MOI.NumberOfConstraints{MOI.ScalarAffineFunction{T}, MOI.LessThan{T}}) where {T, S}
+    return 1
+end
+
+function MOI.get(b::SemiToBinaryBridge{T, S},
+    ::MOI.ListOfConstraintIndices{MOI.SingleVariable, MOI.ZeroOne}) where {T, S}
+    return [b.binary_constraint_index]
+end
+
+function MOI.get(b::SemiToBinaryBridge{T, S},
+    ::MOI.ListOfConstraintIndices{MOI.SingleVariable, MOI.Integer}) where {T, S<:MOI.Semiinteger}
+    return [b.integer_index]
+end
+
+function MOI.get(b::SemiToBinaryBridge{T, S},
+    ::MOI.ListOfConstraintIndices{MOI.ScalarAffineFunction{T}, MOI.LessThan{T}}) where {T, S}
+    return [b.upper_bound_index]
+end
+
+function MOI.get(b::SemiToBinaryBridge{T, S},
+    ::MOI.ListOfConstraintIndices{MOI.ScalarAffineFunction{T}, MOI.GreaterThan{T}}) where {T, S}
+    return [b.lower_bound_index]
+end
+
+function MOI.supports(
+    ::MOI.ModelLike,
+    ::MOI.ConstraintPrimalStart,
+    ::Type{<:SemiToBinaryBridge})
+    return true
+end
+
+function MOI.get(model::MOI.ModelLike, attr::MOI.ConstraintPrimalStart, bridge::SemiToBinaryBridge)
```

Comment:
Missing getter for `ConstraintPrimal`

Reply:
just out of order, I will reorder the functions